pkg/virt-controller/watch/vmi.go

```diff
@@ -353,6 +353,67 @@ func (c *VMIController) execute(key string) error {
 	return nil
 }
 
+// These "dynamic" labels are Pod labels which may diverge from the VMI over time that we want to keep in sync.
+func (c *VMIController) syncDynamicLabelsToPod(pod *k8sv1.Pod, vmi *virtv1.VirtualMachineInstance) error {
+	var patchOps []string
+
+	dynamicLabels := []string{
+		virtv1.NodeNameLabel,
+		virtv1.OutdatedLauncherImageLabel,
+	}
+
+	newMeta := pod.ObjectMeta.DeepCopy()
+	if newMeta.Labels == nil {
+		newMeta.Labels = map[string]string{}
+	}
+
+	changed := false
+	for _, key := range dynamicLabels {
+		val, exists := vmi.Labels[key]
+		podVal, podLabelExists := newMeta.Labels[key]
```

Comment:
nit: let's make them symmetrical.
```suggestion
		vmiVal, vmiLabelExists := vmi.Labels[key]
		podVal, podLabelExists := newMeta.Labels[key]
```